analytics-core/src/main/java/com/segment/analytics/AnalyticsContext.java

```diff
@@ -214,9 +226,10 @@ public Campaign campaign() {
   }
 
   /** Fill this instance with device info from the provided {@link Context}. */
-  void putDevice(Context context) {
+  void putDevice(Context context, boolean collectDeviceID) {
     Device device = new Device();
-    device.put(Device.DEVICE_ID_KEY, getDeviceId(context));
+    String identifier = collectDeviceID ? getDeviceId(context) : traits().anonymousId();
```

Comment:
We could cache it, or alternatively, just leave it empty rather than make up something?